Hw01 hello otus

Sekaiichi opened the pull request:
## Домашнее задание №1 «Hello, OTUS!»

### Чек-лист студента ([Что это?](https://github.com/OtusGolang/home_work/wiki/%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8-%D0%BA-%D1%87%D0%B5%D0%BA-%D0%BB%D0%B8%D1%81%D1%82%D1%83-%D1%81%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0))
- [x] Я перечитал задание после решения. [Зачем перечитывать задание?](https://github.com/OtusGolang/home_work/wiki/%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8-%D0%BA-%D1%87%D0%B5%D0%BA-%D0%BB%D0%B8%D1%81%D1%82%D1%83-%D1%81%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0#user-content-%D0%97%D0%B0%D1%87%D0%B5%D0%BC-%D0%BF%D0%B5%D1%80%D0%B5%D1%87%D0%B8%D1%82%D1%8B%D0%B2%D0%B0%D1%82%D1%8C-%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5)
- [x] Я запустил `go mod tidy`.
- [x] Я удалил `.sync` файл. [Зачем его удалять?](https://github.com/OtusGolang/home_work/wiki/%5B%D0%A1%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0%D0%BC%5D-%D0%9F%D1%80%D0%BE%D1%86%D0%B5%D1%81%D1%81-%D1%81%D0%B4%D0%B0%D1%87%D0%B8-%D0%94%D0%97#user-content-%D0%92%D0%B0%D1%80%D0%B8%D0%B0%D0%BD%D1%82-2)
- [x] Я использовал `stringutil.Reverse`.
- [ ] Я сравнил объем кода с объемом авторского решения. [Зачем сверять объем кода?](https://github.com/OtusGolang/home_work/wiki/%D0%9A%D0%BE%D0%BC%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%80%D0%B8%D0%B8-%D0%BA-%D1%87%D0%B5%D0%BA-%D0%BB%D0%B8%D1%81%D1%82%D1%83-%D1%81%D1%82%D1%83%D0%B4%D0%B5%D0%BD%D1%82%D0%B0#user-content-%D0%97%D0%B0%D1%87%D0%B5%D0%BC-%D1%81%D0%B2%D0%B5%D1%80%D1%8F%D1%82%D1%8C-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BC-%D0%BA%D0%BE%D0%B4%D0%B0)
  - main.go ~12 строк

### Критерии оценки
- [ ] Пайплайн зелёный - 4 балла
- [ ] Используется `stringutil` - 4 балла
- [ ] Понятность и чистота кода - до 2 баллов

#### Зачёт от 7 баллов

Sekaiichi committed:
HW1 is completed

Sekaiichi committed:
HW1 is completed

Sekaiichi committed:
Delete unused .sync file

The .sync file was removed as it is no longer needed in the project. This helps clean up unnecessary files and maintain a tidy codebase.

Sekaiichi committed:
Add reverseText function and its test

Introduced a new reverseText function to replace direct usage of reverse.String in main. Added unit tests to verify the correctness of reverseText functionality.

engineer-go commented on hw01_hello_otus/go.mod:
```diff
@@ -1,3 +1,5 @@
 module github.com/fixme_my_friend/hw01_hello_otus
```
необходимо поправить путь к модулю. Пусть состоит из 1) имени пользователя 2) названия репозитория 3) названия папки в которой хранятся файлы модуля

Sekaiichi replied:
done

Sekaiichi committed:
Update module path in go.mod for repository relocation

The module path was changed to reflect the new repository location. This ensures proper module resolution and avoids potential import issues.
